### PR TITLE
Switch to Tox, add flake8, and fix asserts that would always pass

### PR DESCRIPTION
**Pull request by davehunt**
Once this is merged we'll need to change Jenkins to install tox and run the tests using it. It would also be great to get the tests and flake8 running in Travis to catch any issues coming in from pull requests, but I've left that out for now because there's already a Travis config for Go.

We should also be able to configure requires.io once this lands to highlight any outdated packages, and the `tox.ini` is a default target for the service.

@m8ttyB r?

### CODE REVIEW

**Comment by davehunt** [comment]
Ooh, there's a CircleCI build.. I wonder if we could even run the tests against that?

**Comment by davehunt** [comment]
> We should also be able to configure requires.io once this lands

We'll need a mozilla-services admin to set this up (I am not one).

**Comment by m8ttyB** [comment]
Adding @oremj to the thread.

File: tests/e2e/tests/test_redirects.py [hunk 1]
```diff
@@ -173,23 +173,23 @@ def test_that_checks_redirect_using_incorrect_query_values(self, base_url):
         }
         response = self._head_request(base_url, params=param)
 
-        assert (requests.codes.not_found == response.status_code,
-                self.response_info_failure_message(base_url, param, response))
+        assert requests.codes.not_found != response.status_code, \
```
Comment (m8ttyB) [inline tests/e2e/tests/test_redirects.py]: :p good catch

File: tests/e2e/tests/test_redirects.py [hunk 2]
```diff
@@ -235,21 +235,21 @@ def test_stub_installer_redirect_for_en_us_and_win(self, base_url, product):
...
-                'Failed by redirected to incorrect host. %s' %
-                self.response_info_failure_message(base_url, param, response))
+        assert 'download-installer.cdn.mozilla.net' == parsed_url.netloc, \
+            'Failed by redirected to incorrect host. %s' % \
+            self.response_info_failure_message(base_url, param, response)
 
-        assert ('Unknown' != self.get_x_backend_server(response),
-                'Failed, x-backend-server was not in the response object %s' %
-                self.response_info_failure_message(base_url, param, response))
+        assert 'Unknown' != self.get_x_backend_server(response), \
```
Comment (m8ttyB) [inline tests/e2e/tests/test_redirects.py]: We can remove this assert, `x-backend-server` is no longer something that is in the response object (thx @oremj for weighing in on this).

File: tests/e2e/tests/test_redirects.py (hunk 1 above)
Comment (m8ttyB) [inline tests/e2e/tests/test_redirects.py]: I misread which test this was in relation to, a `404` response is expected.

Comment (davehunt) [inline tests/e2e/tests/test_redirects.py]: oops, this was debugging to make sure a failure reported correctly - thanks for catching this (wish we had Travis CI integration already).

**Comment by davehunt** [comment]
@m8ttyB I've fixed my leftover debug and removed the x-backend-server header check as suggested. There are currently 89 failures.

**Comment by m8ttyB** [comment]
@davehunt this is an awesome bit of work. Thank you bringing the tox to the party.

r+

**Comment by m8ttyB** [comment]
I should also note, through investigation that I was able to reduce the number of failures to only a few that need to be investigated by updating the tests to look for more recent aliases. For example `firefox-33.0` is no longer in use and we should be using `firefox-47.0.1`. 

We can take care of this in a future pr.

**Comment by davehunt** [comment]
I think we should take care of the failures here, unless you're okay with us having these failures appearing in Jenkins. The only reason they're not in Jenkins right now is because they're masked by always asserting a truthy tuple instead of the result of the comparison.

**Comment by m8ttyB** [comment]
Here are the final bits that we can twiddle to fix the test suite are. I'm still looking into what appears to be sporadic failures caused by `Error message: HTTPConnectionPool`:

* Update all instances of `firefox-33.0` to 'firefox-47.0.1'
* Remove checks for a backend server, `assert 'Unknown' != self.get_x_backend_server(response), /`
* From the `LOCALES` [variable](https://github.com/mozilla-services/go-bouncer/pull/28/files#diff-b829282d511ec3ac554b4696d4ed2d8cR12) remove the following; csb, zu, ku

**Comment by davehunt** [comment]
> Remove checks for a backend server, `assert 'Unknown' != self.get_x_backend_server(response)`

Already removed in the PR - reduced failures from 447 to 89.

> Update all instances of `firefox-31.0` to `firefox-47.0.1`

This _increased_ failures from 89 to 441.

> From the `LOCALES` variable remove the following; csb, zu, ku

Reduced failures from 441 to 426 (we removed 15 tests).

@m8ttyB: I've pushed the changes despite the increase in failures - let me know what you'd like me to do next.

**Comment by m8ttyB** [comment]
I ran the tests and saw similar results; stage had ~400 failures, whereas prod exhibited failures of a different nature; 14 that were cause by `Error message: HTTPConnectionPool`

@oremj was able to quickly identify what's going on in with stage. The database hasn't been sync'd against prod in awhile. He's offered to remedy that when he's got a moment.

**Comment by m8ttyB** [comment]
Stage db has been sync'd and the tests look good to me. I'm still seeing some errant `HTTPConnectionPool` failures but I don't believe it's related to your pr.

r+

**Comment by davehunt** [comment]
Thanks @m8ttyB they all pass for me now. I just pushed a small improvement. Let me know if you're going to merge this (I don't have that power) and I can set up Jenkins for it.

**Comment by m8ttyB** [comment]
Looks great. r+ 💯

**Pull request merged**